src/prometheus/definition.go

```diff
@@ -540,6 +558,82 @@ func FromSummary(key string) definition.FetchFunc {
 	}
 }
 
+// FromValueWithLabelsFilter creates a FetchFunc that fetches values from prometheus metrics values given specific
+// labels filter.
+func FromValueWithLabelsFilter(metricName string, nameOverride string, labelsFilter ...LabelsFilter) definition.FetchFunc {
```

Comment:
is `nameOverride` needed here? It might be but since we are already defining the metric name in [Name](https://github.com/newrelic/nri-kubernetes/pull/507/files#diff-a0716046905e3ee93ccaf0bc65ae874ed8d9edda79c4d8a3bc394d57bad9bbe6R63) and also we are not planning to add any suffix it confuses me.

Reply:
AFAIK, it seems necessary. `metricName` in a form like `apiserver_current_inflight_requests` is used to fetch the metric from raw metrics but then we need to override metric name as we expect to have it in a form like `apiserverCurrentInflightRequestsMutating`. Does it make sense to you?

Reply:
could you add an example on the description, I think that will help us to understand what the function does.